lib/form/js/validation.js

```diff
@@ -6,90 +6,91 @@
   'use strict';
 
   var valueCheck = function valueCheck(input, settings, validation) {
-    if (input.target.required && (input.target.required === 'save' || input.target.required === 'publish') && input.target.value === '') {
-      return 'Field cannot be left blank!';
-    }
-
-    return validation(input, settings);
-  };
-
-  // Validation Function
-  // Disabled one-var for better readability
-  var validate = function validate(e) { // eslint-disable-line one-var
-    var target = e.target,
-        value = target.value,
-        inp = allInputs[target.id],
-        parent = allIDs[inp.parent.id],
-        input = {},
-        settings = {},
-        validation = false,
-        message,
-        currentMessage = parent.querySelector('[role="alert"][for="' + target.id + '"]'),
-        insertRef;
+        if (input.target.required && (input.target.required === 'save' || input.target.required === 'publish') && input.target.value === '') {
+          return 'Field cannot be left blank!';
+        }
 
-    // Set up validation input
-    input.target = {
-      value: value,
-      name: inp.name,
-      required: inp.required,
-    };
-    input.all = {};
-    input.all[inp.name] = value;
+        return validation(input, settings);
+      },
 
-    // Set up validation settings
-    settings.target = inp.settings;
-    settings.all = {};
-    settings.all[inp.name] = inp.settings;
+      // Validation Function
+      validate = function validate(e) {
+        var target = e.target,
+            value = target.value,
+            inp = allInputs[target.id],
+            parent = target.closest('div.form--repeatable, div.form--field, fieldset.form--fieldset'),
```

Comment:
And these selectors are over-qualified with the `div`s, should only be the class